tests/integration_test.rs

```diff
@@ -451,26 +452,38 @@ fn test_more_complex_program_with_conditionals() {
         .lt_of_flag(false)
         .build();
     let (result, _final_vm_state) = run_program_with_custom_state(&bin_path, vm_with_custom_flags);
-    dbg!(_final_vm_state);
     assert_eq!(result, U256::from_dec_str("10").unwrap());
 }
 #[test]
 // This test should run out of gas before
 // the program can save a number 3 into the storage.
 fn test_runs_out_of_gas_and_stops() {
     let bin_path = make_bin_path_asm("add_with_costs");
-    let vm = VMStateBuilder::new().gas_left(5510).build();
-    let (result, _) = run_program_with_custom_state(&bin_path, vm);
+    let program_code = program_from_file(&bin_path);
+    let frame = CallFrame::new(program_code, 5510);
```

Comment:
The idea of creating a custom `CallFrame` here is to manually set up the gas, right? The idea for the future is to always use `load_program(...)` function and let the VM generate the new context.

Reply:
Yup, ideally we should always use load_program.